sotodlib/core/metadata/obsdb.py

```diff
@@ -92,39 +92,63 @@ def __init__(self, map_file=None, init_db=True, wafer_info=None):
                     changes = True
             if changes:
                 self.conn.commit()
-        self.primary_keys = self.get_primary_fields()
+        self.primary_keys = self.get_primary_fields(wafer_info)
 
-    def get_primary_fields(self):
+    def get_primary_fields(self, wafer_info=None):
         """Retrieve the primary keys of the specified table.
            This is used whether to index by obs_id or
            obs_id plus additional fields defined by wafer_info."""
         query = "PRAGMA table_info('obs')"
         c = self.conn.execute(query)
         primary_keys = [row['name'] for row in c.fetchall() if row['pk'] > 0]
+        if wafer_info:
+            pkeys = ["obs_id"]
+            pkeys.extend([f"{k}" for k in wafer_info])
+            if sorted(pkeys) != sorted(primary_keys): # sorted allows for different order
+                raise ValueError(f"Primary keys do not match: {primary_keys} != {pkeys}"+
+                                f" must use `wafer_info`=={primary_keys} or create a new dB with {pkeys}")
         return primary_keys
     
-    def warn_primary_keys(self, wafer_info):
+    def _convert_wafer_info(self, obs_id, wafer_info):
+        """Helper function to allow flexibility in way obs_id and wafer_info are passed in."""
+        if isinstance(wafer_info, dict):
+            wafer_info = wafer_info.values()
```

Comment:
I think you need to grab these by key -- they might not be in the order you expect.